lib/chef/resource/cron_access.rb

```diff
@@ -76,9 +81,14 @@ class CronAccess < Chef::Resource
 
       action :deny do
         description "Add the user to the cron.deny file."
+        deny_path = value_for_platform_family(
+            'aix' => '/var/adm/cron/cron.deny',
+            'solaris' => '/etc/cron.d/cron.deny',
+            'default' => '/etc/cron.deny'
```

Comment:
this hash is really crying out for being made a constant and the duplication removed.

Reply:
which duplication do you mean?

Reply:
The arch to path mapping is repeated. That could be extracted into something like this:

```
CRON_PATHS = {           
  'aix' => '/var/adm/cron/',
  'solaris' => '/etc/cron.d/',
  'default' => '/etc/'
}

# later on ...

allow_path = File.join(value_for_platform_family(CRON_PATHS), "cron.allow")

# even later ...

deny_path = File.join(value_for_platform_family(CRON_PATHS), "cron.deny")
```

Reply:
Thank you @phiggins for the explanation! I updated the file.